src/main/scala/is/hail/rvd/OrderedRVDPartitioner.scala

```diff
@@ -18,8 +18,6 @@ class OrderedRVDPartitioner(
   val pkIntervalType = TInterval(pkType)
   val rangeBoundsType = TArray(pkIntervalType)
 
-  assert(rangeBoundsType.typeCheck(rangeBounds))
```

Comment:
This seems like an OK thing to check?

Reply:
I think it's typeCheck that's broken in this case, because `interval<struct{a}>` should let `interval<struct{a, b}>` through. Patrick has been thinking about this more.

Reply:
Hmm. Maybe not typeCheck itself, but perhaps there should be a different way to check this relationship exists, and, if necessary, what code is needed to bridge the physical implementation and the actual implementation.

Reply:
I'm not sure we want to change the behavior of typeCheck, but there is a more relaxed check we could make here.

Reply:
Ok. what we need here is a `TBaseStruct.isComparableAt` (naming is hard) that we can use something like:
```scala
  assert(rangeBounds.forall { case Interval(x,y,_,_) =>
    pkType.isComparableAt(x) && pkType.isComparableAt(y)
  }
```
It should mirror the implementation of `ExtendedOrdering.rowOrdering`, checking that at least the first `pkType.fields.length` worth of columns type check against the corresponding fields.